packages/headless/src/integration-tests/commerce.test.ts

```diff
@@ -0,0 +1,146 @@
+import {buildCart, buildCommerceEngine, buildProductListing} from '../commerce.index';
+import {getOrganizationEndpoints} from '../insight.index';
+import {buildContext} from '../controllers/commerce/context/headless-context';
+import {waitForNextStateChange} from '../test/functional-test-utils';
+
+let accessToken = 'no'
+
+describe('commerce', () => {
+  it("'s working great",
+    async () => {
+      const engine = buildCommerceEngine({
+        configuration: {
+          organizationId: 'commercestore',
+          accessToken,
+          organizationEndpoints: {
+            ...getOrganizationEndpoints('commercestore', 'dev'),
+            platform: 'https://platformdev.cloud.coveo.com',
+          }
+        },
+        //loggerOptions: {level: 'silent'},
+      })
+
+      buildContext(engine, {
+        options: {
+          trackingId: 'commercestore-tracking-id',
+          language: 'en',
+          currency: 'USD',
+          clientId: '41915baa-621c-4408-b9c0-6e59b3cde129',
+          view: {
+            url: 'http://mystore.com/sales'
+          }
+        }
+      })
+
+      const cart = buildCart(engine)
+      cart.addItem({
+        productId: 'nice shoes',
+        quantity: 2
+      })
+      cart.addItem({
+        productId: 'nice shoes',
+        quantity: 3
+      })
+
+      let productListing = buildProductListing(engine);
+      await waitForNextStateChange(engine, {
+        action: () => {
+          productListing.refresh()
+        },
+        expectedSubscriberCalls: 2,
+      });
+
+      expect(productListing.state.products).toEqual([
```

Comment:
Instead of that kind of `strict-equal` on every single property, you can leverage jest to do partial object match.

Also, seems like a lot of these values are ... temporary ? 🤔 ie: will it start to break builds if the underlying org commerce org changes ?

Reply:
Agreed! Just wanted a quick and dirty way of validating that the end-to-end commerce service call worked. Will adjust the test in consequence to use a valid token, use a different org, and better match against expected results.

Reply:
You can `.skip(...)` it in the meantime, if it's not ready.